base/src/main/scala/co/uproot/abandon/Config.scala

```diff
@@ -406,7 +407,52 @@ case class BalanceReportSettings(
   showZeroAmountAccounts: Boolean) extends ReportSettings(_title, _accountMatch, _outFiles) {
 }
 
-case class RegisterReportSettings(_title: String, _accountMatch: Option[Seq[String]], _outFiles: Seq[String]) extends ReportSettings(_title, _accountMatch, _outFiles) {
+/**
+  * GroupBy selectors
+  */
+sealed trait GroupBy
+sealed case class GroupByYear() extends GroupBy
+sealed case class GroupByMonth() extends GroupBy
+sealed case class GroupByDay() extends GroupBy
+sealed case class GroupByIsoWeek() extends GroupBy
+sealed case class GroupByIsoWeekDate() extends GroupBy
+
+object GroupBy {
+  def apply(groupBy: Option[String]): GroupBy = {
+    groupBy match {
+      case Some("year") => GroupByYear()
+      case Some("month") => GroupByMonth()
+      case Some("day") => GroupByDay()
+      case Some("isoWeek") => GroupByIsoWeek()
+      case Some("isoWeekDate") => GroupByIsoWeekDate()
+      /* default */
+      case None => GroupByMonth()
+      /* Error*/
+      case _ => throw new SettingsError(
+        "GroupBy: unknown operator. Valid operators are: year, month, day, isoWeek, isoWeekDate")
+    }
+  }
+}
+
+case class RegisterReportSettings(_title: String,  _accountMatch: Option[Seq[String]], _outFiles: Seq[String], groupBy: GroupBy) extends ReportSettings(_title, _accountMatch, _outFiles) {
```

Comment:
minor: private vals are being prefixed with `_`. It would be good to rename `groupBy` to `_groupBy` just for convention sake.